cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,496 @@
+					vec![(
+						RococoRelayReceiver::get(),
+						Data::Raw(vec![1_u8; 1].try_into().unwrap())
+					)],
+				));
+			},
+			Subs::Many(n) => {
+				let mut subs = Vec::new();
+				for i in 0..n {
+					subs.push((
+						RococoRelayReceiver::get(),
+						Data::Raw(vec![i as u8; 1].try_into().unwrap()),
+					));
+				}
+				assert_ok!(RococoIdentity::set_subs(
+					RococoOrigin::signed(RococoRelaySender::get()),
+					subs,
+				));
+			},
+		}
+
+		let reserved_bal = RococoBalances::reserved_balance(RococoRelaySender::get());
+		let id_deposit = id_deposit_relaychain(&id.relay.main);
+
+		match id.relay.subs {
+			Subs::Zero => {},
+			Subs::One => {
+				total_deposit = SubAccountDeposit::get() + id_deposit_relaychain(&id.relay.main);
+				// for Many and One we assert the same
+				assert_expected_events!(
+					RococoRelay,
+					vec![
+						RuntimeEvent::Identity(IdentityEvent::IdentitySet { .. }) => {},
+						RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+							who: *who == RococoRelaySender::get(),
+							amount: *amount == id_deposit,
+						},
+						RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+							who: *who == RococoRelaySender::get(),
+							amount: *amount == SubAccountDeposit::get(),
+						},
+					]
+				);
+				// The reserved balance should equal the calculated total deposit
+				assert_eq!(reserved_bal, total_deposit);
+			},
+			Subs::Many(n) => {
+				let mut sub_account_deposit = 0_u128;
+				for _ in 0..n {
+					sub_account_deposit += SubAccountDeposit::get();
+				}
+				total_deposit = sub_account_deposit + id_deposit_relaychain(&id.relay.main);
+				assert_expected_events!(
+					RococoRelay,
+					vec![
+						RuntimeEvent::Identity(IdentityEvent::IdentitySet { .. }) => {},
+						RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+							who: *who == RococoRelaySender::get(),
+							amount: *amount == id_deposit,
+						},
+						RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+							who: *who == RococoRelaySender::get(),
+							amount: *amount == sub_account_deposit,
+						},
+					]
+				);
+				// The reserved balance should equal the calculated total deposit
+				assert_eq!(reserved_bal, total_deposit);
+			},
+		}
+	});
+	total_deposit
+}
+
+fn assert_set_id_parachain(id: &Identity) {
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		let free_bal = PeopleRococoBalances::free_balance(PeopleRococoSender::get());
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		assert_ok!(PeopleRococoIdentity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			id.para.main.clone(),
+		));
+
+		assert_ok!(PeopleRococoIdentity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
```

Comment:
You need to call it `n` times.

Reply:
I created a `fn set_subs_no_deposit(who: &T::AccountId, subs: Vec<T::AccountId>)` in the Identity pallet, as using `set_sub_no_deposit` was overwriting the value for the same key when calling it n times. 

With this new method, it's working and the expected events are triggering. 

We may want to remove `set_sub_no_deposit` and always work work `Vec<T::AccountId>`. What do you think?